workers/proxycacheblobworker.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import time
+
+import features
+from app import app, proxy_cache_blob_queue
+from data.database import ImageStorage, ImageStoragePlacement, ManifestBlob
+from data.model import repository, user
+from data.registry_model.datatypes import RepositoryReference
+from data.registry_model.registry_proxy_model import ProxyModel
+from util.log import logfile_path
+from workers.gunicorn_worker import GunicornWorker
+from workers.queueworker import (
+    JobException,
+    QueueWorker,
+    WorkerSleepException,
+    WorkerUnhealthyException,
+)
+from workers.worker import with_exponential_backoff
+
+logger = logging.getLogger(__name__)
+
+
+POLL_PERIOD_SECONDS = 10
+RESERVATION_SECONDS = 60 * 20
+
+
+class ProxyCacheBlobWorker(QueueWorker):
+    def process_queue_item(self, job_details):
+        repo_id = job_details["repo_id"]
+        namespace_name = job_details["namespace"]
+        digest = job_details["digest"]
+        username = job_details["username"]
+
+        repo_name = repository.lookup_repository(repo_id).name
+        repo_ref = RepositoryReference.for_id(repo_id)
+        user_ref = user.get_username(username)
+
+        registry_proxy_model = ProxyModel(
+            namespace_name,
+            repo_name,
+            user_ref,
+        )
+
+        logger.debug(
+            "Starting proxy cache blob download for digest %s for repo id %s",
+            digest,
+            repo_id,
+        )
+
+        blob = registry_proxy_model._get_shared_storage(digest)
```

Comment:
i'm assuming this is just to check for the global empty layer? otherwise this check looks redundant to line 53